Code quality & documentation

ChrisAxt opened the pull request:
Code quality improvements as well as code documentation, most changes were merged into master through the Arduino_Unittesting due to compatibility issues.  

resolves #83 
closes #83

ChrisAxt committed:
Improving code quality and documentation #83
##########################################################
Tags
 #83
Changes
- Added code documentation to DriverLogin.java, PassengerDashboard.java and ThumbstickView.java

ChrisAxt committed:
Merge remote-tracking branch 'origin/master' into Code_quality_&_documentation

# Conflicts:
#	UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java

ChrisAxt committed:
Improving code quality and documentation #83
##########################################################
Merged master into branch to keep updated

ChrisAxt committed:
Merge remote-tracking branch 'origin/master' into Code_quality_&_documentation

# Conflicts:
#	UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java

ChrisAxt committed:
Improving code quality and documentation #83
##########################################################
Merged master into branch to keep updated

Commit:
Merge remote-tracking branch 'origin/master' into Code_quality_&_documentation

# Conflicts:
#	UI/app/src/main/java/com/example/pathfinder/Activities/DriverDashboard.java
#	UI/app/src/main/java/com/example/pathfinder/Activities/PassengerDashboard.java

alrapal reviewed (approved):
Thanks for the documentation ! Makes it way easier to follow. 
As you said the PR #96 implemented a lot of your changes, and I saw that the switch was actually from this branch ! As we say in french "Rendre à César ce qui appartient à César" ! 
(Give back to Cesar what belongs to Cesar). So good work ! 😄

alrapal commented on UI/app/src/main/java/com/example/pathfinder/Activities/DriverLogin.java:
```diff
@@ -100,15 +99,25 @@ public void onClick(View v) {
+    /**
+     * Method to check if a given name exists in the system
+     * @param username
+     * @return
```
Could we add just what is return here ?

iachella reviewed (approved):
Looks good! Nice way of documenting :)

alrapal reviewed (commented):
(Not sure it is necessary but what do you think about the comments below. I know we can look at the code to know it, but I think it makes it neater to put it in the return parameter

alrapal commented on UI/app/src/main/java/com/example/pathfinder/Activities/DriverLogin.java:
```diff
@@ -100,15 +99,25 @@ public void onClick(View v) {
+     * Method to check if a given username and password belong to the same account
+     * @param username
+     * @param password
+     * @return
```
And here ?

ChrisAxt replied:
Good catch, will make the changes :)

alrapal commented on UI/app/src/main/java/com/example/pathfinder/Activities/ThumbstickView.java:
```diff
@@ -107,10 +130,19 @@ public void surfaceDestroyed(@NonNull SurfaceHolder holder) {
+     * Method that calculates in which position the thumbstick currently is
+     * @param v
+     * @param e
+     * @return
```
here as well?

alrapal reviewed (commented):
(Just pointing out where we could add explanation regarding what is returned and when)

juliavankirk commented:
Looks great, thanks for working on this!

ediz-genc commented:
> Code quality improvements as well as code documentation, most changes were merged into master through the Arduino_Unittesting due to compatibility issues.

bassamEldesouki commented:
Seems like everything is covered now, no comment from me, Thanks for this work :)

alrapal commented:
Reopen because needs to be merged and addressed

Commit:
Code quality & documentation #98
#########################################################
Changes
- Changes made in documentation for return value descriptions on methods that didn't include any.

ediz-genc reviewed (approved):
Other than tiny changes, all documentation looks good. Good work :)

alrapal reviewed (approved):
Thanks for the changes ! 😄
